message_ix_models/tests/model/material/test_build.py

```diff
@@ -0,0 +1,58 @@
+import logging
+
+import pytest
+
+from message_ix_models.model.material import build
+from message_ix_models.model.structure import get_codes
+from message_ix_models.testing import bare_res
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "regions_arg, regions_exp",
+    [
+        # ("R11", "R11"),
+        ("R12", "R12"),
+    ],
+)
+@pytest.mark.parametrize("material", [None])
+def test_make_spec(regions_arg, regions_exp, material):
+    # The spec can be generated
+    spec = build.make_spec(regions_arg, material)
+
+    # The required elements of the "node" set match the configuration
+    nodes = get_codes(f"node/{regions_exp}")
+    expected = list(map(str, nodes[nodes.index("World")].child))
+    assert expected == spec["require"].set["node"]
+
+
+@pytest.mark.parametrize(
+    "regions, years, relations, solve",
+    [
+        ("R12", "B", "B", False),
+        pytest.param(
+            "R11", "B", "B", False, marks=pytest.mark.xfail(raises=NotImplementedError)
```

Comment:
Is this the line that changed from my last review?
If yes, it is probably the cause of the CI failure.
The `test_create_res` in `tests/model/test_bare` expects certain models to have specific numbers of `sets`, `relations`, etc defined. My guess is: this test increases the number of `relations` defined on a `bare_res R11` scenario from 20 to 319 before it runs into the `NotImplementedError` and thus the other test fails.

There are a few options here and maybe @khaeru can jump in to clarify what is most desirable: I see that this PR has already adapted the expected number of `sets` for the other test. The way we currently handle the `bare_res` scenario, this seems to be fine, but I'm wondering if we weren't better off with `bare_res` being a fixture and providing just clones of itself? 
That way, we wouldn't need to adapt the `test_create_res`, which shouldn't have anything to do with `materials` and could just use the scenario.
Alternatively, you could continue adapting that test's expectations so long as it doesn't cause other tests to fail.

Reply:
Sounds great to me :)

Reply:
Thanks for jumping in here, @khaeru. I was suspecting some kind of side effect coming from my most recent test addition, but I am not familiar enough with the test suite to quickly figure that out on my own. Please go ahead and push your suggested changes.

Reply:
Done, let's see how it works.